python/taichi/types/ndarray_type.py

```diff
@@ -9,78 +12,107 @@ def __init__(self, element_type, shape=None):
         self.layout = Layout.AOS
 
 
+# TODO(Haidong): This is a helper function that creates a MatrixType
+#                with respect to element_dim and element_shape.
+#                Remove this function when the two args are totally deprecated.
+def make_matrix_dtype_from_element_shape(element_dim, element_shape,
+                                         primitive_dtype):
+    if isinstance(primitive_dtype, CompoundType):
+        raise TypeError(
+            f'Cannot specifiy matrix dtype "{primitive_dtype}" and element shape or dim at the same time.'
+        )
+
+    # Scalars
+    if element_dim == 0 or (element_shape is not None
+                            and len(element_shape) == 0):
+        return primitive_dtype
+
+    # Cook element dim and shape into matrix type.
+    mat_dtype = None
+    if element_dim is not None:
+        if element_dim < 0 or element_dim > 2:
```

Comment:
we may need to support dtype ndim > 2 in the future but a TODO comment is enough if we don't have a use case for now.